tests/cases/compiler/reverseMappedTypeInferFromItsProperties.ts

```diff
@@ -0,0 +1,53 @@
+// @strict: true
+// @noEmit: true
+
+// repro from #29479
+
+type Selector<S, R> = (state: S) => R;
+
+declare function createStructuredSelector<S, T>(selectors: {
+  [K in keyof T]: Selector<S, T[K]>;
+}): Selector<S, T>;
+
+type State = { foo: number };
+
+declare const mySelector: Selector<State, boolean>;
+
+export const result = createStructuredSelector({ mySelector });
+
+// 52737#discussion_r1127035242
+declare const otherSelectors: { [x: string]: Selector<State, boolean> };
+export const otherResult = createStructuredSelector(otherSelectors);
+
+declare function inferFromValue<T, S extends string>(obj: { [K in keyof T]: S }): [T, S];
+const fromValue1 = inferFromValue({ a: "foo", b: "bar" });
+
+// 52737#issuecomment-1457046429
```

Comment:
this isn't yet covered by this PR, it's the case found [here](https://github.com/microsoft/TypeScript/pull/52737#issuecomment-1457046429)